### ISSUE TITLE
[BUG] Providing maxSize validation on NonEmptyList does not get applied in OpenAPI

### ISSUE THREAD

**Issue by matwojcik** [softwaremill/tapir#1047]
Tapir version: 0.17.12

Scala version: 2.13.5

**Describe the bug**

When I try to add some validation on a schema with `NonEmptyList` using  https://tapir.softwaremill.com/en/latest/endpoint/customtypes.html#customising-derived-schemas it seems to get ignored.

**How to reproduce?**
```scala

import sttp.tapir._
import sttp.tapir.generic.auto._
import sttp.tapir.integ.cats.codec._

case class Wrapper(items: NonEmptyList[String])

implicit val customWrapperSchema: Schema[Wrapper] = implicitly[Derived[Schema[Wrapper]]]
  .value
  .modify(_.items)(_.validate(Validator.maxSize[String, List](1).contramap(_.toList)).description("Some description"))

val yaml = OpenAPIDocsInterpreter.toOpenAPI(endpoint.out(jsonBody[Wrapper]), Info("Fruits", "1.0")).toYaml
```

the max size validation gets ignored:

```
 schemas:
    Wrapper:
      required:
      - items
      type: object
      properties:
        items:
          type: array
          items:
            type: string
          description: Some description
          minItems: 1
```

**Comment by Qthunder** [comment]
I think I may have hit the same issue. See my gitter message `https://gitter.im/softwaremill/tapir?at=603910cbb5131f4f28db2266`

The solution might be very simple - waiting for an answer to see what the original intention might have been.

**Comment by matwojcik** [comment]
@adamw ok, it is now in openapi, but is not validated by server - I thought it should result in 400, but it passes the invalid body

**Comment by adamw** [comment]
@matwojcik ah yes ... I was aware that this design has limitations and planned a rework but now there's even more reason to do this :) coming soon :) reopening then

**Comment by adamw** [comment]
Should be properly fixed now in 0.18.0-M1 :)
